src/browser/ui/dom/components/ReactDOMInput.js

```diff
@@ -115,6 +138,24 @@ var ReactDOMInput = ReactClass.createClass({
   _handleChange: function(event) {
     var returnValue;
     var onChange = LinkedValueUtils.getOnChange(this.props);
+
+    // IE 10+ fire input events when setting/unsetting a placeholder
+    // we guard against it by checking if the next and
+    // last values are both empty and bailing out of the change
+    // https://github.com/facebook/react/issues/3484
+    if (isIEInputEvent(event)) {
+      var controlledValue = LinkedValueUtils.getValue(this.props);
+      var lastValue = controlledValue != null ?
+        '' + controlledValue :
+        this._uncontrolledValue;
+
+      this._uncontrolledValue = event.target.value;
+
+      if ( event.target.value === '' && lastValue === '') {
+        return returnValue;
```

Comment:
This seems fragile considering there's lots of code beneath this that may or may not need to be executed, when all this really wants to do is discard the call to `onChange` right?